### PR TITLE
Additional board hooks

### PR DESCRIPTION
**Pull request by dragomano**
Added integrate_load_board, integrate_board_info, integrate_pre_boardindex and integrate_boardindex_board hooks

Signed-off-by: Bugo <bugo@dragomano.ru>

### CODE REVIEW

File: Sources/Subs-BoardIndex.php [hunk 1]
```diff
@@ -26,106 +26,113 @@
  * array with categories->boards->child_boards or an associative array
  * with boards->child_boards.
  *
- * @param array $boardIndexOptions An array of boardindex options
+ * @param array $board_index_options An array of boardindex options
  * @return array An array of information for displaying the boardindex
  */
 
-function getBoardIndex($boardIndexOptions)
+function getBoardIndex($board_index_options)
```
Comment (live627) [inline Sources/Subs-BoardIndex.php:33]: curious as to what prompted this change

Comment (dragomano) [reply Sources/Subs-BoardIndex.php:33]: Just a small analogy with the $message_index_selects in MessageIndex.php :)

File: Sources/Load.php [hunk 2]
```diff
@@ -951,11 +957,12 @@ function loadBoard()
 				LEFT JOIN {db_prefix}moderators AS mods ON (mods.id_board = {raw:board_link})
 				LEFT JOIN {db_prefix}members AS mem ON (mem.id_member = mods.id_member)
 			WHERE b.id_board = {raw:board_link}',
-			array(
+			array_merge(array(
 				'current_topic' => $topic,
```
Comment (MissAllSunday) [inline Sources/Load.php]: Might be good to declare $custom_column_parameters with both keys current_topic and board_link so mod authors can make use of them, even change them if they want to.

Comment (dragomano) [reply Sources/Load.php]: Done

**Review by MissAllSunday — APPROVED** [review]

**Pull request merged**